### PR TITLE
JIT: encapsulate general checks for optimization

### PR DESCRIPTION
**Pull request by AndyAyersMS**
Add methods that answer the general question of whether or not
the jit is optimizing the code it produces.

Use this to replace composite checks for minopts and debug
codegen (the two modes where the jit is not optimizing).

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
Follow up from #21792 (and some other places where I said I'd do this).

No diffs in pmi for normal jitting or for minopts; no diffs for crossgen.

There are still a number of places that only check either for debug or for minopts that probably deserve closer scrutiny.

cc @dotnet/jit-contrib

File: src/jit/compiler.h [hunk 1]
```diff
@@ -8279,6 +8279,16 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
             return compMinOptsIsSet;
         }
 #endif // !DEBUG
+
+        inline bool OptimizationDisabled()
```
Comment (mikedn) [inline src/jit/compiler.h]: The inline specifier isn't needed on function definitions that are inside the class.

Comment (AndyAyersMS) [reply src/jit/compiler.h]: Scrubbed these and moved the `LclVarDsc` constructor into the class.

**Comment by AndyAyersMS** [comment]
Test failures all seem unrelated. Will rerun a few...

@dotnet-bot retest Ubuntu x64 Checked Innerloop Build and Test
@dotnet-bot retest Windows_NT arm Cross Checked Innerloop Build and Test

**Pull request merged**